crypto/src/test_utils/zero_crypto.rs

```diff
@@ -0,0 +1,400 @@
+    fn eq(&self, _other: &Self) -> bool {
+        true
+    }
+}
+
+impl Eq for ZeroSignature {}
+
+impl Signature for ZeroSignature {
+    fn from_bytes(_bytes: &[u8]) -> Result<Self, signature::Error> {
+        Ok(ZeroSignature {})
+    }
+}
+
+impl Display for ZeroSignature {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> Result<(), fmt::Error> {
+        write!(f, "{}", Base64::encode_string(self.as_ref()))
+    }
+}
+
+impl Authenticator for ZeroSignature {
+    type PubKey = ZeroPublicKey;
+    type PrivKey = ZeroPrivateKey;
+    const LENGTH: usize = 0;
+}
+
+///
+/// Implement SigningKey
+///
+
+impl AsRef<[u8]> for ZeroPrivateKey {
+    fn as_ref(&self) -> &[u8] {
+        &self.0
+    }
+}
+
+impl ToFromBytes for ZeroPrivateKey {
+    fn from_bytes(bytes: &[u8]) -> Result<Self, signature::Error> {
+        let bytes: [u8; PRIVATE_KEY_LENGTH] =
+            bytes.try_into().map_err(|_| signature::Error::new())?;
+        Ok(ZeroPrivateKey(bytes))
+    }
+}
+
+// There is a strong requirement for this specific impl. in Fab benchmarks
+impl Serialize for ZeroPrivateKey {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        serializer.serialize_str(&self.encode_base64())
+    }
+}
+
+// There is a strong requirement for this specific impl. in Fab benchmarks
+impl<'de> Deserialize<'de> for ZeroPrivateKey {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: de::Deserializer<'de>,
+    {
+        let s = <String as serde::Deserialize>::deserialize(deserializer)?;
+        let value = Self::decode_base64(&s).map_err(|e| de::Error::custom(e.to_string()))?;
+        Ok(value)
+    }
+}
+
+impl SigningKey for ZeroPrivateKey {
+    type PubKey = ZeroPublicKey;
+    type Sig = ZeroSignature;
+    const LENGTH: usize = PRIVATE_KEY_LENGTH;
+}
+
+impl Signer<ZeroSignature> for ZeroPrivateKey {
+    fn try_sign(&self, _msg: &[u8]) -> Result<ZeroSignature, signature::Error> {
+        Ok(ZeroSignature {})
+    }
+}
+
+///
+/// Implement KeyPair
+///
+
+impl From<ZeroPrivateKey> for ZeroKeyPair {
+    fn from(secret: ZeroPrivateKey) -> Self {
+        let pk: ZeroPublicKey = (&secret).into();
+        ZeroKeyPair { name: pk, secret }
+    }
+}
+
+impl EncodeDecodeBase64 for ZeroKeyPair {
+    fn decode_base64(value: &str) -> Result<Self, eyre::Report> {
+        keypair_decode_base64(value)
+    }
+
+    fn encode_base64(&self) -> String {
+        let mut bytes: Vec<u8> = Vec::new();
+        bytes.extend_from_slice(self.secret.as_ref());
+        bytes.extend_from_slice(self.name.as_ref());
+        base64ct::Base64::encode_string(&bytes[..])
+    }
+}
+
+impl KeyPair for ZeroKeyPair {
+    type PubKey = ZeroPublicKey;
+    type PrivKey = ZeroPrivateKey;
+    type Sig = ZeroSignature;
+
+    #[cfg(feature = "copy_key")]
+    fn copy(&self) -> Self {
+        ZeroKeyPair {
+            name: ZeroPublicKey(self.name.0),
```

Comment:
One day, not in this PR, we should replace `.name` with a meaningful value (i.e., `public_key`, `private_key`)

Reply:
Will open an issue on this